tools/aiecc/aiecc.cpp

```diff
@@ -0,0 +1,862 @@
+#include <iostream>
+#include <string>
+#include <regex>
+#include <stdlib.h>
+#include <filesystem>
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Program.h"
+#include "llvm/Support/Path.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/OwningOpRef.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Support/FileUtilities.h"
+#include "llvm/Support/ToolOutputFile.h"
+#include "llvm/Support/SourceMgr.h"
+#include "aie/InitialAllDialect.h"
+#include "mlir/InitAllDialects.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/InitAllPasses.h"
+#include "mlir/Dialect/Affine/Passes.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "aie/Dialect/AIE/Transforms/AIEPasses.h"
+#include "aie/Dialect/AIEX/Transforms/AIEXPasses.h"
+#include "mlir/Tools/mlir-translate/Translation.h"
+#include "aie/Targets/AIETargets.h"
+#include "mlir/Target/LLVMIR/Export.h"
+#include "mlir/Target/LLVMIR/Dialect/Builtin/BuiltinToLLVMIRTranslation.h"
+#include "mlir/Target/LLVMIR/Dialect/LLVMIR/LLVMToLLVMIRTranslation.h"
+#include "configure.h"
+#include "llvm/Support/JSON.h"
+
+using namespace llvm;
+using namespace mlir;
+namespace sfs = std::filesystem;
+
+cl::opt<std::string> FileName(cl::Positional, cl::desc("<input mlir>"), cl::Required);
+cl::opt<std::string> TmpDir("tmpdir", cl::desc("Directory used for temporary file storage"));
+cl::opt<std::string> SysRoot("sysroot", cl::desc("sysroot for cross-compilation"));
```

Comment:
All of these need to be put under a `cl::opt::Category` so they pollute the args list. [Random example from upstream](https://github.com/llvm/llvm-project/blob/ea668144d9c6d93fbe2ea7827bd8e8cbaf4c50dd/llvm/tools/llvm-reduce/ReducerWorkItem.cpp#L51-L54).